x/farming/keeper/reward_test.go

```diff
@@ -176,3 +176,28 @@ func (suite *KeeperTestSuite) TestDistributeRewards() {
 	suite.Require().True(found)
 	suite.Require().Equal(suite.ctx.BlockTime(), lastDistributedAt)
 }
+
+func (suite *KeeperTestSuite) TestHarvest() {
+	for _, plan := range suite.samplePlans {
+		suite.keeper.SetPlan(suite.ctx, plan)
+	}
+
+	suite.Stake(suite.addrs[0], sdk.NewCoins(sdk.NewInt64Coin(denom1, 1_000_000)))
+	suite.keeper.ProcessQueuedCoins(suite.ctx)
+
+	balancesBefore := suite.app.BankKeeper.GetAllBalances(suite.ctx, suite.addrs[0])
+
+	suite.ctx = suite.ctx.WithBlockTime(mustParseRFC3339("2021-08-05T00:00:00Z"))
```

Comment:
I think this block time and `suite.samplePlans`'s `StartTime` and `EndTime` are depending on when the test is run, there is a risk of having different results. It needs to be a clear past date and a future date long after IMO.

Reply:
Hm, as far as I know `StartTime` and `EndTime` of each `suite.samplePlans` are hard-coded times so these don't depend on current time.